processing/src/main/java/org/apache/druid/query/groupby/GroupingEngine.java

```diff
@@ -686,13 +686,27 @@ public Sequence<ResultRow> processSubtotalsSpec(
           processingConfig.intermediateComputeSizeBytes()
       );
 
-      List<String> queryDimNames = baseSubtotalQuery.getDimensions().stream().map(DimensionSpec::getOutputName)
+      List<String> queryDimNamesInOrder = baseSubtotalQuery.getDimensions().stream().map(DimensionSpec::getOutputName)
```

Comment:
I think I understand the reasons/etc here - but if these are the column the query's output will be ordered; can't we place these logics inside a member method as `GroupByQuery#getOrderColumnNames` or something ?